spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -311,6 +325,210 @@ def _measure_axial_damage_ratio(self, im_data, p_lst, idx):
+                # Get the indices of the lesion mask for the selected axial slice
+                lesion_indices = np.where(slice_lesion_data)[0]
+                # Get the indices of the spinal cord mask for the selected axial slice
+                sc_indices = np.where(slice_sc_data)[0]
+
+                # Compute ventral and dorsal tissue bridges
+                dorsal_bridge_width = lesion_indices[0] - sc_indices[0]         # [0] returns the most dorsal elements
+                # if the lesion extends the spinal cord, the dorsal bridge is set to 0
+                if dorsal_bridge_width < 0:
+                    dorsal_bridge_width = 0
+                ventral_bridge_width = sc_indices[-1] - lesion_indices[-1]      # [-1] returns the most ventral elements
+                # if the lesion extends the spinal cord, the ventral bridge is set to 0
+                if ventral_bridge_width < 0:
+                    ventral_bridge_width = 0
+
+                tissue_bridges_dict[sagittal_slice, axial_slice] = \
+                    {'dorsal_bridge_width': dorsal_bridge_width,
+                     'ventral_bridge_width': ventral_bridge_width,
+                     'lesion_indices': lesion_indices}       # we store lesion indices for plotting
+
+        # --------------------------------------
+        # Get minimal tissue bridges
+        # --------------------------------------
+        # Convert the dictionary to a DataFrame (for easier manipulation)
+        # 1. Create a MultiIndex from the dictionary keys
+        index = pd.MultiIndex.from_tuples(tissue_bridges_dict.keys(), names=['sagittal_slice', 'axial_slice'])
+        # 2. Create the DataFrame using the MultiIndex and the dictionary values
+        tissue_bridges_df = pd.DataFrame(list(tissue_bridges_dict.values()), index=index)
+        # 3. Reset the index to make 'sagittal_slice' and 'axial_slice' as columns
+        tissue_bridges_df.reset_index(inplace=True)
+
+        # Initialize auxiliary dict to store info for plotting
+        plot_dict = {}
+        # Get slices of minimum dorsal and ventral tissue bridges for each sagittal slice
+        # NOTE: we get minimum because tissue bridges are quantified as the width of spared tissue at the minimum
+        # distance from cerebrospinal fluid to the lesion boundary
+        for sagittal_slice in sagittal_lesion_slices:
+            # Get df for the selected sagittal slice
+            df_temp = tissue_bridges_df[tissue_bridges_df['sagittal_slice'] == sagittal_slice]
+            # Get the axial slice with the minimum dorsal tissue bridge for the selected sagittal slice
+            min_dorsal_bridge_width_slice = df_temp.loc[df_temp['dorsal_bridge_width'].idxmin(), 'axial_slice']
+            # Get the axial slice with the minimum ventral tissue bridge for the selected sagittal slice
+            min_ventral_bridge_width_slice = df_temp.loc[df_temp['ventral_bridge_width'].idxmin(), 'axial_slice']
```

Comment:
I absolutely love your use of comments in most of this method. They really helped me understand the functionality!

I think you may be able to trim down on some of the comments, though? I like to have 1 comment per "block" of related lines. For example, in this case, you're finding the minimum for both ventral/dorsal. So, you could do this instead:

```suggestion
            # Get the axial slices corresponding to the minimum bridge widths
            min_dorsal_bridge_width_slice = df_temp.loc[df_temp['dorsal_bridge_width'].idxmin(), 'axial_slice']
            min_ventral_bridge_width_slice = df_temp.loc[df_temp['ventral_bridge_width'].idxmin(), 'axial_slice']
```

That way, you still get the gist of your comments, and also the single comment helps to visually group the related chunks of code.

(There are a lot of parts of the code that have 1 comment per line -- I think you could probably simplify most of these into 1 comment per "block of lines".)

Reply:
Thank you! Absolutely agree!